# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/thebes/gfxFontEntry.cpp
# Commit: 095b3edec3c8
# Full Hash: 095b3edec3c8ba9318c71d0acc91e10ce57818dc
# Author: Jonathan Kew <jkew@mozilla.com>
# Date: 2019-04-29 21:53:38
# Regressor Bug: 1514869
# File Overlap Count: 1
# Description:
#   Bug 1514869 - patch 2 - Adapt platform-font-list code to work with either the existing in-process font list or cross-process shared font list. r=jwatt
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D22938
# ==============================================================================

diff -r 21ef00977ab6 -r 095b3edec3c8 gfx/thebes/gfxFontEntry.cpp
--- a/gfx/thebes/gfxFontEntry.cpp	Sat Apr 27 15:37:29 2019 +0000
+++ b/gfx/thebes/gfxFontEntry.cpp	Sat Apr 27 15:37:58 2019 +0000
@@ -141,11 +141,13 @@
 }
 
 bool gfxFontEntry::TestCharacterMap(uint32_t aCh) {
-  if (!mCharacterMap) {
+  if (!mCharacterMap && !mShmemCharacterMap) {
     ReadCMAP();
-    NS_ASSERTION(mCharacterMap, "failed to initialize character map");
+    MOZ_ASSERT(mCharacterMap || mShmemCharacterMap,
+               "failed to initialize character map");
   }
-  return mCharacterMap->test(aCh);
+  return mShmemCharacterMap ? mShmemCharacterMap->test(aCh)
+                            : mCharacterMap->test(aCh);
 }
 
 nsresult gfxFontEntry::InitializeUVSMap() {
@@ -1694,8 +1696,9 @@
                                   FontInfoData* aFontInfoData) {
   // if all needed names have already been read, skip
   if (mOtherFamilyNamesInitialized &&
-      (mFaceNamesInitialized || !aNeedFullnamePostscriptNames))
+      (mFaceNamesInitialized || !aNeedFullnamePostscriptNames)) {
     return;
+  }
 
   bool asyncFontLoaderDisabled = false;
 